### ISSUE TITLE
Generic type alias that includes a Callable results in false positives

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#1497]
The following code produces an error when it shoudln't.

```python
from typing import Callable, TypeVar

T = TypeVar("T")
F = Callable[[T], T]

def f() -> F[T]:
    def g(x: T) -> T:
        ...
    return g

g = f()
x = g("foo")
```

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.112, which I just published. It will also be included in the next release of Pylance.
